modules/gapi/src/backends/cpu/gparsers.cpp

```diff
@@ -0,0 +1,332 @@
+    {
+        auto h_scale = in_sz.height;
+        auto w_scale = in_sz.width;
+        cv::Rect r;
+        r.x = static_cast<int>((x - w / 2) * w_scale);
+        r.y = static_cast<int>((y - h / 2) * h_scale);
+        r.width = static_cast<int>(w * w_scale);
+        r.height = static_cast<int>(h * h_scale);
+        return r;
+    }
+
+private:
+    const float* m_out;
+    int m_side, m_lcoords, m_lclasses;
+
+    int index(const int& i, const int& b, const int& entry)
+    {
+        return b * m_side * m_side * (m_lcoords + m_lclasses + 1) + entry * m_side * m_side + i;
+    }
+};
+
+struct YoloParams
+{
+    int    num = 5;
+    int coords = 4;
+};
+
+class SSDParser
+{
+public:
+    SSDParser(const cv::MatSize& in_ssd_dims, const cv::Size& in_size, const float* const data)
+        : m_dims(in_ssd_dims), m_maxProp(in_ssd_dims[2]), m_objSize(in_ssd_dims[3]),
+          m_data(data), m_surface(cv::Rect({0,0}, in_size)), m_size(in_size)
+    {
+        GAPI_Assert(in_ssd_dims.dims() == 4u); // Fixed output layout
+        GAPI_Assert(m_objSize  == 7);          // Fixed SSD object size
+    }
+
+    void adjustBoundingBox(cv::Rect& boundingBox)
+    {
+        auto w = boundingBox.width;
+        auto h = boundingBox.height;
+
+        boundingBox.x -= static_cast<int>(0.067 * w);
+        boundingBox.y -= static_cast<int>(0.028 * h);
+
+        boundingBox.width += static_cast<int>(0.15 * w);
+        boundingBox.height += static_cast<int>(0.13 * h);
+
+        if (boundingBox.width < boundingBox.height)
+        {
+            auto dx = (boundingBox.height - boundingBox.width);
+            boundingBox.x -= dx / 2;
+            boundingBox.width += dx;
+        }
+        else
+        {
+            auto dy = (boundingBox.width - boundingBox.height);
+            boundingBox.y -= dy / 2;
+            boundingBox.height += dy;
+        }
+    }
+
+    std::tuple<cv::Rect, float, float, int> unitedParse(const size_t& step)
+    {
+        const float* it = m_data + step * m_objSize;
+        float image_id   = it[0];
+        int   label      = static_cast<int>(it[1]);
+        float confidence = it[2];
+        float rc_left    = it[3];
+        float rc_top     = it[4];
+        float rc_right   = it[5];
+        float rc_bottom  = it[6];
+
+        cv::Rect rc;  // Map relative coordinates to the original image scale
+        rc.x      = static_cast<int>(rc_left   * m_size.width);
+        rc.y      = static_cast<int>(rc_top    * m_size.height);
+        rc.width  = static_cast<int>(rc_right  * m_size.width)  - rc.x;
+        rc.height = static_cast<int>(rc_bottom * m_size.height) - rc.y;
+        return std::make_tuple(rc, image_id, confidence, label);
+    }
+
+    int getMaxProposals()
+    {
+        return m_maxProp;
+    }
+
+    cv::Rect getSurface()
+    {
+        return m_surface;
+    }
+
+private:
+    const cv::MatSize m_dims;
+    int m_maxProp = 0, m_objSize = 0;
+    const float* m_data;
```

Comment:
I suggest to explicitly initialize a pointer by `nullptr`, and cv::Matsize (if it is enum) by some `int` value

Reply:
Fixed. cv::Matsize is struct.